### PR TITLE
Add subtree filter option for codec service. Use subtree for crud service read

### PR DESCRIPTION
**Pull request by abhikeshav**
* rename path::CodeService --> path::Codec
 * XML Subtree codec
 * Move cpp bundles away from ydk:: namespace to individual bundle namespace
  like openconfig::
 * Update documentation/add tests to support subtree filter codec
 * Change crud read to use subtree codec in creating RPC request for netconf

### CODE REVIEW

File: sdk/python/core/setup.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def build_extension(self, ext):
                                       get_python_library()),
                       '-DPYTHON_INCLUDE={0}'.format(
                                       get_python_include()),
-                      '-DCMAKE_BUILD_TYPE=Release']
+                      '-DCMAKE_BUILD_TYPE=debug']
```
Comment (psykokwak4) [inline sdk/python/core/setup.py]: Should this be `Release`?

File: sdk/python/core/samples/path_bgp.py [hunk 2]
```diff
@@ -34,7 +35,7 @@ def execute_path(provider, codec):
 
 
 if __name__ == "__main__":
-    repo = Repository("/var/folders/jt/kfhqscp54dq3gfwdpkbj48zm0000gn/T//127.0.0.1:12022")
-    provider = RestconfServiceProvider(repo, '127.0.0.1', 'admin', 'admin', 8008)
-    codec = CodecService()
+    repo = Repository("/Users/abhirame/.ydk/127.0.0.1:12022")
```
Comment (psykokwak4) [inline sdk/python/core/samples/path_bgp.py]: Could `expanduser` be utilized in this case?

File: sdk/python/core/setup.py (hunk 1 above)
Comment (abhikeshav) [inline sdk/python/core/setup.py]: Good catch

File: sdk/cpp/core/src/json.hpp [hunk 3]
```diff
@@ -12416,7 +12416,7 @@ class basic_json
                 {
                     // op must be "add", "remove", "replace", "move", "copy", or
                     // "test"
-                    JSON_THROW(std::invalid_argument("yfilter value '" + op + "' is invalid"));
+                    JSON_THROW(std::invalid_argument("operation value '" + op + "' is invalid"));
```
Comment (abhikeshav) [inline sdk/cpp/core/src/json.hpp]: These were unintentionally renamed previously

File: sdk/python/core/samples/path_bgp.py (hunk 2 above)
Comment (abhikeshav) [inline sdk/python/core/samples/path_bgp.py]: Possibly. Maybe I should remove the `repo` and have the sample download the models

**Pull request merged**